File: solutions/fortran/collatz-conjecture/1/collatz_conjecture.f90

module collatz_conjecture
  implicit none
contains

  integer function steps(i)
    integer :: i
    steps = 0
    do while (i>1)
      if (mod(i,2) == 0) then
        i = i / 2
      else
        i = 3 * i + 1
      end if
      steps = steps + 1   
    end do  
    return
  end function

end module

File: solutions/julia/collatz-conjecture/1/collatz-conjecture.jl
function collatz_steps(n)
    s::Int=0
    if n <= 0
        throw(DomainError(n))
    end
    while n>1
        s+=1
        if mod(n,2)==0
            n=n/2
        else
            n=3n+1
        end
    end
    return s
end